### PR TITLE
lint fixes 1 - tests

### PR DESCRIPTION
**Pull request by leonardlin**
lint fixes part 1, tests

### CODE REVIEW

File: test/job.js [hunk 1]
```diff
@@ -765,21 +765,18 @@ describe('Job', () => {
...
-        }, async() => {
+      const processorPromise = new Promise(resolve => {
+        agenda.define('lock job', { lockLifetime: 50 }, () => {
           startCounter++;
 
           if (startCounter !== 1) {
             expect(startCounter).to.be(2);
-            await agenda.stop();
-            resolve();
+            return agenda.stop().then(() => { // eslint-disable-line promise/prefer-await-to-then
```
Comment (simison) [inline test/job.js:774]: How come this needed `.then()`? 🤔 

Would be good to add a code comment to explain lint disable.

You might also shorten it to `.then(resolve)`

**Review by simison — DISMISSED** [review]
Looking good!

Overall I prefer to have a brief comment on why a lint disable was needed for other devs reading the code later on, but not a blocker for merging.

File: test/job.js [hunk 2]
```diff
@@ -794,21 +790,18 @@ describe('Job', () => {
...
-        }, async(job, cb) => { // eslint-disable-line no-unused-vars
+      const processorPromise = new Promise(resolve => {
+        agenda.define('lock job', { lockLifetime: 50 }, (job, cb) => { // eslint-disable-line no-unused-vars
           runCount++;
 
           if (runCount !== 1) {
             expect(runCount).to.be(2);
-            await agenda.stop();
-            resolve();
+            agenda.stop().then(() => { // eslint-disable-line promise/prefer-await-to-then
```
Comment (simison) [inline test/job.js:799]: Another likely not needed `.then()` here.

File: test/job.js (hunk 1 above)
Comment (leonardlin) [reply test/job.js]: changed to await style

File: test/job.js (hunk 2 above)
Comment (leonardlin) [reply test/job.js]: changed to await style

**Review by simison — APPROVED** [review]
Thanks!

**Pull request merged**